Dockerfile

```
@@ -19,166 +20,294 @@
+ARG PYTHON_VIRTUALENV_VERSION=15.1.0
+ARG PIPTOOL_VERSION=21.3.1
+
+ENV PYENV_ROOT=/opt/python
+RUN curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
+ENV PATH=/opt/python/bin:$PATH
+# Python 3.6.x series has problems with alignment with modern GCC
+# As we don not want patch Python versions we use Clang as compiler
+RUN CC=clang-9 CXX=clang++9 pyenv install "${PYTHON_VERSION}"
+RUN pyenv global "${PYTHON_VERSION}"
+ENV PATH=/opt/python/shims:$PATH
+RUN pip install -U \
+    conan=="${CONAN_VERSION}" \
+    pip=="${PIPTOOL_VERSION}" \
+    pipenv \
+    Mercurial \
+    virtualenv=="${PYTHON_VIRTUALENV_VERSION}" \
+    wheel
+
+COPY docker/python.sh /etc/ort/bash_modules
+
+#------------------------------------------------------------------------
+# RUBY - Build Ruby as a separate component with rbenv
+FROM build as rubybuild
+
+ARG RUBY_VERSION=2.7.4
+ENV RBENV_ROOT=/opt/rbenv
+ENV PATH=${RBENV_ROOT}/bin:${RBENV_ROOT}/shims/:${RBENV_ROOT}/plugins/ruby-build/bin:$PATH
+
+RUN git clone --depth 1 https://github.com/rbenv/rbenv.git ${RBENV_ROOT}
+RUN git clone --depth 1 https://github.com/rbenv/ruby-build.git "$(rbenv root)"/plugins/ruby-build
+WORKDIR ${RBENV_ROOT}
+RUN src/configure \
+    && make -C src
+RUN rbenv install ${RUBY_VERSION} \
+    && rbenv global ${RUBY_VERSION} \
+    && gem install bundler cocoapods
+
+# Install 'CocoaPods'. As https://github.com/CocoaPods/CocoaPods/pull/10609 is needed but not yet released.
+RUN curl -ksSL https://github.com/CocoaPods/CocoaPods/archive/9461b346aeb8cba6df71fd4e71661688138ec21b.tar.gz | \
+    tar -zxC . \
+    && (cd CocoaPods-9461b346aeb8cba6df71fd4e71661688138ec21b \
+        && gem build cocoapods.gemspec \
+        && gem install cocoapods-1.10.1.gem \
+        ) \
+    && rm -rf CocoaPods-9461b346aeb8cba6df71fd4e71661688138ec21b
+
+COPY docker/ruby.sh /etc/ort/bash_modules
+
+#------------------------------------------------------------------------
+# NODEJS - Build NodeJS as a separate component with nvm
+FROM build AS nodebuild
+
+ARG BOWER_VERSION=1.8.8
+ARG NODEJS_VERSION="--lts"
+ARG NPM_VERSION=7.20.6
+ARG YARN_VERSION=1.22.10
+
+ENV NVM_DIR=/opt/nodejs
+
+RUN git clone --depth 1 https://github.com/nvm-sh/nvm.git /opt/nodejs
+RUN . $NVM_DIR/nvm.sh && nvm install "${NODEJS_VERSION}"
+RUN . $NVM_DIR/nvm.sh \
+    && npm install --global npm@$NPM_VERSION bower@$BOWER_VERSION yarn@$YARN_VERSION
+COPY docker/nodejs.sh /etc/ort/bash_modules
+
+#------------------------------------------------------------------------
+# GOLANG - Build as a separate component
+FROM build AS gobuild
+
+ARG GO_DEP_VERSION=0.5.4
+ARG GO_VERSION=1.17.3
+ENV GOPATH=/opt/go
+RUN curl -L https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz | tar -C /opt -xz
+ENV PATH=/opt/go/bin:$PATH
+RUN go version
+RUN curl -ksS https://raw.githubusercontent.com/golang/dep/v$GO_DEP_VERSION/install.sh | bash
+RUN echo "add_local_path /opt/go/bin:$PATH" > /etc/ort/bash_modules/go.sh
```

when is the `$PATH` variable resolved? On docker build time? Might need escaping:

```suggestion
RUN echo "add_local_path /opt/go/bin:\$PATH" > /etc/ort/bash_modules/go.sh
```

---

Really liked this approach. Many thanks @maxhbr 
I merged your PR